### ISSUE TITLE
Identity Defined in Augmenting Yang Is Invisible in Startup Config

### ISSUE THREAD

**Issue by haiyunhzhang** [sysrepo/sysrepo#1481]
### Things you may try first
(put "x" in "[]" if you already tried following)
* [x] Did you check if this is a duplicate issue?
* [x] Did you test it on the latest sysrepo devel branch?

### Description
Referenced identities by a startup XML file, defined in the augmenting Yang model are failed to resolve

### Steps to Reproduce
1. Install original Yang model,
2. Install a Yang model, that augments the original Yang.
3. Subscribe the original Yang model
4. Import a startup XML file to the Yang models.
5. The referenced identities by the startup XML file, defined in the augmenting Yang model are failed to resolve. 

**Expected behavior:**
 [What you expected to happen]
The import of the startup XML file should success

**Actual behavior:**
 [What actually happened]
[ERR] Unable to parse the input data: Failed to resolve identityref,

### Versions
Please provide versions for software needed to reproduce the bug:
Sysrepo version: `sysrepod -v`. 
sysrepod - sysrepo daemon, version 0.7.7

libyang version: `yanglint -v`
yanglint 1.1.5

netopeer2-server: `netopeer2-server -V` (optional)
netopeer2-server 0.7.11

### Attachments
Please run sysrepo daemon in debug mode and write the logs to a file. For example:
`sysrepod -d -l 4 &> sysrepod.log`.
No error returned from the log

Similarly, if Netopeer2 server is used, also provide us with its log:
`netopeer2-server -d -v3 &> netopeer2-server.log` (optional)
No log printed

You're also welcomed to provide us with any other data you think may be useful.
For example:
Installed modules: `sysrepoctl -l`
Content of datastore used: check files in  Sysrepo data directory (default `/etc/sysrepo/data`).

Thanks!

**Comment by michalvasko** [comment]
Hi,
could you please provide all the models you are using and the data for them so I can reproduce this problem?

Regards,
Michal

**Comment by haiyunhzhang** [comment]
Hi Michal,

Attached includes all yang and config files. I used the following command to install and config the sample.

sudo sysrepoctl --install --yang=main.yang
sudo sysrepoctl --install --yang=augment.yang

python python_sample_aug.py
sudo sysrepocfg --import=conf.json --format=json main

Because the augment yang module subscription was rejected by sysrepod, the python_sample_aug.py subscribes the main yang module.


sudo sysrepocfg --import=conf.json --format=json main
[ERR] Unable to parse the input data: When condition "derived-from-or-self(main:class, 'augment:AS-SAMPLE')" not satisfied. (/main:config/augment:augment)
Errors were encountered during importing. Cancelling the operation.

If you remove the following when statement from the augment.yang, the error will be gone.

when "derived-from-or-self(main:class, 'augment:AS-SAMPLE')";

There needs a way to make the when condition true.



Regards,

Harry

**Comment by haiyunhzhang** [comment]
Hi Michal,

I think it is my problem. After I added class type, the error was disappeared. I am new to Yang model. After search from some online tutorial materials, I figured it out.
{
   "main:config": {
     "class": "augment:AS-SAMPLE",
     "augment:augment": {
       "test": "test"
     }
   }
}

Thank you so much for quick response. I may come back for help later.


Regards,

Harry



From: Zhang, Harry
Sent: Tuesday, April 30, 2019 12:15 PM
To: 'sysrepo/sysrepo' <reply@reply.github.com>
Subject: RE: [sysrepo/sysrepo] Identity Defined in Augmenting Yang Is Invisible in Startup Config (#1481)

Hi Michal,

Attached includes all yang and config files. I used the following command to install and config the sample.

sudo sysrepoctl --install --yang=main.yang
sudo sysrepoctl --install --yang=augment.yang

python python_sample_aug.py
sudo sysrepocfg --import=conf.json --format=json main

Because the augment yang module subscription was rejected by sysrepod, the python_sample_aug.py subscribes the main yang module.


sudo sysrepocfg --import=conf.json --format=json main
[ERR] Unable to parse the input data: When condition "derived-from-or-self(main:class, 'augment:AS-SAMPLE')" not satisfied. (/main:config/augment:augment)
Errors were encountered during importing. Cancelling the operation.

If you remove the following when statement from the augment.yang, the error will be gone.

when "derived-from-or-self(main:class, 'augment:AS-SAMPLE')";

There needs a way to make the when condition true.



Regards,

Harry
